components/brave_policy/ad_block_only_mode/ad_block_only_mode_policies_utils.cc

```diff
@@ -0,0 +1,55 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_policy/ad_block_only_mode/ad_block_only_mode_policies_utils.h"
+
+#include "base/values.h"
+#include "build/build_config.h"
+#include "components/content_settings/core/common/content_settings.h"
+#include "components/policy/core/common/policy_bundle.h"
+#include "components/policy/core/common/policy_map.h"
+#include "components/policy/core/common/policy_namespace.h"
+#include "components/policy/core/common/policy_types.h"
+#include "components/policy/policy_constants.h"
+
+void LoadAdBlockOnlyModePolicies(policy::PolicyBundle& bundle) {
+  // This guard is necessary because the Ad Block Only mode policies below are
```

Comment:
os guards should only be used for os-specific code, not to guard feature-specific code that doesn't apply to all platforms. I'm not clear on why these need a buildflag guard in the first place, does ios not build if you remove it?

Reply:
why is the provider used at all on ios is none of the policies work on ios

Reply:
neither because this file shouldn't exist, it should be in AdBlockOnlyPolicyManager::GetAllProfilePolicies like brave origin and the entire AdBlockOnlyPolicyManager should be guarded be an ad_block_only_mode feature flag, not ios.

Reply:
what I'm suggesting is exactly what brave origin does. There shouldn't be any components/policy dependencies because those can be handled with interface/impl/forward declarations